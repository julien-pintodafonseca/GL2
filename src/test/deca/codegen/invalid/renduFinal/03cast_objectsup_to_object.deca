// Description:
//      Utilisation du cast
//      => On ne peut pas cast un objet d'une classe mère dans une classe fille.
//
// Resultats:
//    Erreur à l'exécution
//    Ligne 20 : cast invalide

class B {

}

class A extends B {

}

{
    A a = new A();
    B b = new B();
    a = (A)(b);
}
