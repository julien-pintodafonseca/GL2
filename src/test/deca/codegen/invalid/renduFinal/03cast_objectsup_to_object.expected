Erreur: cast impossible de B vers A
